packages/libs/user-datasets/src/lib/StoreModules/UserDatasetListStoreModule.ts

```diff
@@ -98,6 +99,7 @@ export function reduce(state: State = initialState, action: Action): State {
         : state;
 
     case DETAIL_REMOVE_SUCCESS:
+      // @ts-ignore
```

Comment:
I tried to remove this one, and get this error:
```
Type '{ userDatasets: (string | number)[]; userDatasetsById: { [x: string]: { isLoading: false; resource: UserDataset; } | undefined; }; status: "complete"; filterByProject: boolean; }' is not assignable to type 'State'.
  Types of property 'userDatasetsById' are incompatible.
    Type '{ [x: string]: { isLoading: false; resource: UserDataset; } | undefined; }' is not assignable to type 'Record<string, { isLoading: false; resource: UserDataset; }>'.
      'string' index signatures are incompatible.
        Type '{ isLoading: false; resource: UserDataset; } | undefined' is not assignable to type '{ isLoading: false; resource: UserDataset; }'.
          Type 'undefined' is not assignable to type '{ isLoading: false; resource: UserDataset; }'.ts(2322)
```

I think it is because we are setting the value of the removed dataset to `undefined`.

I think this would work, using `omit` from `lodash`:
```ts
      return state.status === 'complete'
        ? {
            ...state,
            userDatasets: difference(state.userDatasets, [
              action.payload.userDataset.id,
            ]),
            userDatasetsById: omit(state.userDatasetsById, action.payload.userDataset.id)
          }
        : state;
```